src/ansys/dpf/core/geometry.py

```diff
@@ -0,0 +1,403 @@
+
+from ansys.dpf import core as dpf
+
+from ansys.dpf.core import Field
+from ansys.dpf.core.fields_factory import field_from_array
+from ansys.dpf.core.plotter import DpfPlotter
+
+import numpy as np
+
+
+def normalize_vector(vector):
+    """Normalize vector."""
+    return vector / np.linalg.norm(vector)
+
+
+class Points:
+    """
+    Collection of DPF points.
+
+    This can be a single point or a cloud of points.
+
+    Parameters
+    ----------
+    coordinates: array, list, Field
+        Coordinates of the points in a 3D space.
+
+    Examples
+    --------
+    Create Points object with two points, print object information and plot.
+
+    >>> from ansys.dpf.core.geometry import Points
+    >>> points = Points([[0, 0, 0], [1, 0, 0]])
+    >>> print(points)
+    DPF Points object:
+    Number of points: 2
+    Coordinates:
+      [[0. 0. 0.]
+      [1. 0. 0.]]
+    >>> points.plot()
+
+    """
+
+    def __init__(self, coordinates, server=None):
+        # Input check
+        if not isinstance(coordinates, Field):
+            coordinates = field_from_array(coordinates, server=server)
+
+        self._coordinates = coordinates
+        self._server = server
+
+    def __getitem__(self, value):
+        return self.coordinates.data[value]
+
+    def __len__(self):
+        return self.num_points
+
+    def __str__(self):
+        """Print Points information."""
+        txt = "DPF Points object:\n"
+        txt += f"Number of points: {self.num_points}\n"
+        txt += f"Coordinates:\n  {self._coordinates.data}\n"
+        return txt
+
+    @property
+    def coordinates(self):
+        """Coordinates of the Points."""
+        return self._coordinates
+
+    @property
+    def num_points(self):
+        """Total number of points."""
+        return (
+            self._coordinates.shape[0]
+            if isinstance(self._coordinates.shape, tuple)
+            else 1
+        )
+
+    @property
+    def dimension(self):
+        """Dimension of the Points object space."""
+        return 3
+
+    def plot(self, **kwargs):
+        """Visualize Points object."""
+        pl = DpfPlotter(**kwargs)
+        pl.add_points(self._coordinates.data)
+        pl.show_figure(show_axes=True)
+
+
+class Line:
+    """
+    Create Line object from two 3D points.
+
+    Parameters
+    ----------
+    coordinates: array, list, Field, Points
+        3D coordinates of the two points defining the line.
+    num_points: int
+        Number of points used to discretize the line.
+
+    Examples
+    --------
+    Create a line from two points, print object information and plot.
+
+    >>> from ansys.dpf.core.geometry import Line
+    >>> line = Line([[0, 0, 0], [1, 0, 0]])
+    >>> print(line)
+    DPF Line object:
+    Starting point: [0. 0. 0.]
+    Ending point: [1. 0. 0.]
+    Line discretized with 100 points
+    >>> line.plot()
+
+    """
+
+    def __init__(self, coordinates, num_points=100, server=None):
```

Comment:
I dont like to have ``num_points`` hardcoded.... Also I dont see why we need to have the line discretized. I think it is for the intersection with other objects, but I would expect pyvista would take care of that.

Reply:
I agree with doing the intersection before.

If it is for mapping, I'm fine with that.